actions/utils.go

```diff
@@ -46,3 +56,30 @@ func join(A []string, delim string) string {
 
 	return buffer.String()
 }
+
+// Randomly generate a set of indexes in each sector
+func generateInsertedIndexesForPearl(fileSizeInByte int) []int {
+	var indexes []int
+	if fileSizeInByte <= 0 {
+		return indexes
+	}
+
+	fileSectorInByte := fileChunkSizeInByte * (fileSectorInChunkSize - 1)
+	numOfSectors := int(math.Ceil(float64(fileSizeInByte) / float64(fileSectorInByte)))
+	remainderOfChunks := int(math.Ceil(float64(fileSizeInByte) / fileChunkSizeInByte))
+
+	for i := 0; i < numOfSectors; i++ {
+		rang := min(fileSectorInChunkSize, remainderOfChunks)
```

Comment:
I think you will have an off by 1 error. You need to add 1 to the remainderOfChunks since the treasure chunk will extend the last sector.

Reply:
When I compute the remainderOfChunk, I offset that by 1.

Reply:
In line 74